python/CMakeLists.txt

```diff
@@ -18,6 +18,56 @@ if(MATERIALX_PYTHON_OCIO_DIR)
   install(DIRECTORY "${MATERIALX_PYTHON_OCIO_DIR}/" DESTINATION "${MATERIALX_PYTHON_FOLDER_NAME}/config/" MESSAGE_NEVER)
 endif()
 
+if (MATERIALX_PYTHON_STUBS)
+  # attempts to find mypy stubgen executable
+  find_program(_MYPY_STUBGEN_EXECUTABLE "stubgen")
+  message("Found stubgen executable: _MYPY_STUBGEN_EXECUTABLE=${MATERIALX_PYTHON_STUBS}")
+
+  # check if stubgen can be accessed
+  execute_process(
+    COMMAND ${_MYPY_STUBGEN_EXECUTABLE} --help
+    OUTPUT_VARIABLE _MYPY_STUBGEN_HELP_OUTPUT
+    ERROR_VARIABLE _MYPY_STUBGEN_HELP_OUTPUT
+    RESULT_VARIABLE _MYPY_STUBGEN_HELP_EXITCODE
+    COMMAND_ECHO STDOUT
+  )
+
+  # build stubs if stubgen could be found properly
+  if (_MYPY_STUBGEN_HELP_EXITCODE EQUAL "0")
+    message("mypy stubgen found successfully, setting up Python Stubs build process...")
+    install(CODE "
+      message(\"Building Python Stubs...\")
+      message(\"MATERIALX_PYTHON_FOLDER_NAME=${MATERIALX_PYTHON_FOLDER_NAME}\")
+      message(\"CMAKE_INSTALL_PREFIX=${CMAKE_INSTALL_PREFIX}\")
+      if(EXISTS ${_MYPY_STUBGEN_EXECUTABLE})
+        execute_process(
+          COMMAND ${_MYPY_STUBGEN_EXECUTABLE} -p MaterialX --inspect-mode -o ${CMAKE_INSTALL_PREFIX}/python -v --ignore-errors
```

Comment:
I'm not sure if `CMAKE_INSTALL_PREFIX` will point at the right thing. See https://scikit-build-core.readthedocs.io/en/latest/cmakelists.html#install-directories.